### PR TITLE
Aggregate localizable guides for 1.16-2.2 to 2.3

### PR DESCRIPTION
**Pull request by tnir**
Blocked by #624

### What was the end-user problem that led to this PR?

While guides are versioned with URLs, they have not been changed since Bundler v1.16.

### What was your diagnosis of the problem?

Guides were duplicated via `rake versions:create`.

### What is your fix for the problem, implemented in this PR?

- Existing duplicated codes are removed.
- New duplication will stop.

### Why did you choose this fix out of the possible options?

`https://bundler.io/v2.3/guides/creating_gem.html` should be now `https://bundler.io/guides/creating_gem.html` as we do not have to version this guide in fact.

### CODE REVIEW

File: source/partials/_guides_sidebar.haml [hunk 1]
```diff
@@ -1,6 +1,9 @@
+- @guides = guides
 %ul
-  - guides.each do |page|
+  - @guides.each do |page|
     %li{class: current_guide?(page[:filename])}
       = link_to_guide(page)
     %li.separator
       %hr
+  - if @guides.size == 0
+    %li ❗There is any guide for this Bundler version any more. Check out guides for the latest version.
```
Comment (deivid-rodriguez) [inline source/partials/_guides_sidebar.haml:9]: So if I understand correctly, now the sidebar with the list of guides is only displayed for 1.15 and 2.3, and the rest show this alert?

I actually think we can take this a bit further. How about:

* Make guides completely version independent. If the only difference between them is usage of modern hashes, I think we can add a note at the beginning that the guide does not work on old Rubies due to this, and that users of these Rubies should upgrade. From now on, if we introduced something version dependent to any guide, we can add a note about the version where the feature described was first introduced.

* Replace the sidebar for each "what's new page" to instead show the documentation sidebar for each version (the one [here](https://bundler.io/v2.3/man/bundle-install.1.html)).

* Keep the guide list only at https://bundler.io/docs.html, which is version independent, and redirect all versioned urls to the non-versioned urls.

How about that?

Comment (tnir) [reply source/partials/_guides_sidebar.haml:9]: Totally agree wit you, especially first and second ones.

> Keep the guide list only at https://bundler.io/docs.html, which is version independent, and redirect all versioned urls to the non-versioned urls.

Per last one, it depends on traffic (page views) of `/docs.html`. Weeks ago I found https://bundler.io/doc/policies.html, which has no link from. If `/docs.html` has less traffic like `/doc/policies.html`, that would be a living dead...

Comment (deivid-rodriguez) [reply source/partials/_guides_sidebar.haml:9]: Can you answer

> So if I understand correctly, now the sidebar with the list of guides is only displayed for 1.15 and 2.3, and the rest show this alert?

If this is correct, I think for now, before we implement the other steps, I'd rather keep the list of links but change them to be redirects to either the 1.15 guide or the 2.3 guide accordingly.

Comment (tnir) [reply source/partials/_guides_sidebar.haml:9]: > So if I understand correctly, now the sidebar with the list of guides is only displayed for 1.15 and 2.3, and the rest show this alert?

Yes. But,

> [...] I'd rather keep the list of links but change them to be redirects to either the 1.15 guide or the 2.3 guide accordingly.

I do not think it is easy to do without an additional workaround for version conversion from 1.12-1.14 and 1.16-2.2 to 1.15 and 2.3.

File: config.rb [hunk 2]
```diff
@@ -93,6 +93,14 @@
   end
 end
 
+# Redirect old localizable guides (v1.16-v2.2) to the latest version (v2.3) of guide compatible
+["", "pl/"].each do |lang|
+  %w[1.16 1.17 2.0 2.1 2.2].each do |version|
+    redirect "#{lang}v#{version}/guides/creating_gem.html", to: "#{lang}v2.3/guides/creating_gem.html"
+    redirect "#{lang}v#{version}/guides/using_bundler_in_applications.html", to: "#{lang}v2.3/using_bundler_in_applications.html"
```
Comment (deivid-rodriguez) [inline config.rb:100]: This is introducing the same bug fixed by #642.

File: helpers/guides_helper.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ module GuidesHelper
 
   def guides
     guides = Dir.glob("./source/#{current_visible_version}/guides/*")
-    target_version = [current_visible_version, "v1.15"].max
+    target_version = current_visible_version > "v1.15" ? current_version : "v1.15"
```
Comment (deivid-rodriguez) [inline helpers/guides_helper.rb:7]: This is unnecessary, I guess it comes from the rebase.

**Review by deivid-rodriguez — COMMENTED** [review]
It seems another step in the good direction, just left two comments.

File: config.rb (hunk 2 above)
Comment (tnir) [inline config.rb:100]: Analogy to #642, this should be:
```suggestion
    redirect "#{lang}v#{version}/guides/using_bundler_in_applications.html", to: "#{lang}v2.3/guides/using_bundler_in_applications.html"
```

File: helpers/guides_helper.rb (hunk 3 above)
Comment (tnir) [reply helpers/guides_helper.rb:7]: Nope, `current_visible_version` is the version to render in Middleman and `current_version` is always `v2.3` (as of now).

Comment (deivid-rodriguez) [reply helpers/guides_helper.rb:7]: Oh, sorry, my bad, I misread the code. Thanks!

**Review by deivid-rodriguez — APPROVED** [review]
Thanks!

**Pull request merged**